### ISSUE TITLE
listsinceblock (Linux command line) does not show "generate" on address until assigned to account

### ISSUE THREAD

**Issue by Nuwisam** [gridcoin-community/Gridcoin-Research#501]
Recently I staked a block with my "change" address. Unfortunately it does not show in list from "listsinceblock". Interestingly when I receive normal transaction on that address it does show in "listsinceblock".

Is that known / expected behaviour?

**Comment by tomasbrod** [comment]
Does it show in the GUI?

**Comment by Nuwisam** [comment]
Yeah, forgot to mention - I am using Linux command line client (gridcoinresearchd).

**Comment by tomasbrod** [comment]
Are you able to start a GUI client? If you are, then please check if the transactions appear in the gui.

**Comment by Nuwisam** [comment]
Sorry, not really - it is headless server.

**Comment by Nuwisam** [comment]
getbalance is affected by this transaction and it shows correct value.

The same is true for "list rsa"- TxCount, Last Block Paid and CPID Lifetime Research Paid are showing correct values.

Just listsinceblock hides it.

**Comment by iFoggz** [comment]
this rpc call uses same general function as listtransactions

**Comment by Nuwisam** [comment]
Could you please point me the class having this "general function"? Maybe I could take a quick look myself and try to figure it out.

**Comment by Nuwisam** [comment]
So... I assigned an account to this address (by gridcoinresearchd setaccount) and now it shows "generate" transaction in listsinceblock (reminder - it showed "receive" transactions for this address even before). It kinda resolves my issue.

However I still would like to know (if someone has time and desire to explain):
1) Why was the change address used to stake a block? My suspicion - it was once used before for "receive" transaction.
2) Since it was used to stake a block - why did listsinceblock hide it when it had no account / name assigned? Why only "generate" transactions?
3) Is there a way to list all addresses (named / accounted and unnamed / unaccounted) associated with my wallet?
(all referring to Linux command line)

**Comment by tomasbrod** [comment]
1) any P2PKH or P2PK unspent transaction output will be used for staking if it is not reserved by settings, contains non-dust amount and is mature (16h prod, 1h test). You simply sent some money, change returned to you using change address and then it staked.
2) idk
3) listaddressgroupings

**Comment by Nuwisam** [comment]
First of all - thanks for your answer.

Ad 2. - I would love to see if it is expected result or some minor bug, but I am unable to find mentioned "general function" for listing transactions in any class (there are simply to many of them). Maybe if someone points me in the right direction (class name) I would be able to investigate from there.

Ad 3. - I know this command, but it does not fully satisfy my needs, as it lists only "non-empty" addresses. Windows client somehow gets even those without any transactions that are assigned to given wallet (I was using Windows client for a moment and it showed all the addresses generated by "getnewaddress", even those without account / name and transaction).

**Comment by iFoggz** [comment]
ok looking at this. I see that it uses the old `ListTransactions` which does things a lot different. Turns out that `ListTransactions` function alone does not show change period even for transactions. I had sent some coins to an address and let the change go to a change address yet it would not show in `listsinceblock` at all. The `ListTransactions` alone does show generated transactions however not for change either.

I've since tested using `ListTransactions2` here and it now shows change addresses as well. so i'll test a Stake with a change address to see if this is included. if so a migration to `ListTransactions2` here may be the solution
